### ISSUE TITLE
OpenApiSpecGeneratorPlugin snippet is missing configSection

### ISSUE THREAD

**Issue by waldekmastykarz** [garrytrinder/dev-proxy-toolkit#160]
https://github.com/garrytrinder/dev-proxy-toolkit/blob/167dff4fe66cdc613db78b48fe7fd619d4cd275d/src/snippets.json#L783-L793

We have a separate config snippet, but the plugin instance misses the `configSection` property that links them both.

- [x] #172
- [x] #173

**Comment by garrytrinder** [comment]
The `configSection` is missing in this snippet as it's optional.

For plugins where the `configSection` is mandatory we include it in the snippet and the extension will show an error until a `configSection` is created that matches the name given.

![image](https://github.com/user-attachments/assets/f4984a2b-2a01-4db6-966b-961a7c3c0b7a)

That said, maybe we should think of a way to highlight that a plugin can have configuration even if it is optional. Thoughts?

**Comment by waldekmastykarz** [comment]
I found it confusing, because we allow you to add a plugin instance and its config but they aren't tied together which is easy to overlook without proxy throwing a single error or warning. It would improve the user experience if it was clearer when:

1. A plugin supports a config, albeit optionally
2. You added a config but it's not tied to the plugin instance

Let's think what's the best way to do it.

**Comment by garrytrinder** [comment]
> 1. A plugin supports a config, albeit optionally

Agreed. One way in which we can demonstrate this is by adding an information diagnostic to the plugin name value. This will bring your attention to it when the plugin is added.

> 2. You added a config but it's not tied to the plugin instance

As configuration is just a named object, we will need to make the assumption that an object is a configuration section and check to see if any plugin has a configSection with that name. We currently check if a plugin has a configSection but not the other way around.

**Comment by waldekmastykarz** [comment]
> Agreed. One way in which we can demonstrate this is by adding an information diagnostic to the plugin name value. This will bring your attention to it when the plugin is added.

How would that look like?

> As configuration is just a named object, we will need to make the assumption that an object is a configuration section and check to see if any plugin has a configSection with that name. We currently check if a plugin has a configSection but not the other way around.

Good idea and I think it's safe for us to make that assumption, at least for now. Let's try it!

**Comment by garrytrinder** [comment]
> How would that look like?

Something like this... (ignore the text)

![image](https://github.com/user-attachments/assets/e5d6fcc4-0225-40c6-93da-e32feafb58a7)

**Comment by waldekmastykarz** [comment]
Let's try it!

**Comment by garrytrinder** [comment]
- [ ] Add information diagnostic when a plugin has optional configSection
- [ ] Add warning diagnostic when a configSection exists but is not connected to a plugin

**Comment by garrytrinder** [comment]
Thanks for the feedback @waldekmastykarz 

https://github.com/garrytrinder/dev-proxy-toolkit/issues/172 and https://github.com/garrytrinder/dev-proxy-toolkit/issues/173 have been merged and will be available in the next pre-release version.

**Comment by waldekmastykarz** [comment]
Cool! Looking forward to trying it out!
